Add an error message for unknown ROS distributions.

clalancette opened the pull request:
This adds a 404 for unknown ROS distributions rather than a
page that just doesn't work.

Fixes https://github.com/ros-infrastructure/prerelease_website/issues/22

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

clalancette committed:
Add an error message for unknown ROS distributions.

This adds a 404 for unknown ROS distributions rather than a
page that just doesn't work.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

mikaelarguedas commented:
While we don;t ship dry distros anymore, this is specific to wet distros making it impossible to have dry_distros on the prerelease website if someone wants right ?

clalancette commented:
> While we don;t ship dry distros anymore, this is specific to wet distros making it impossible to have dry_distros on the prerelease website if someone wants right ?

I guess that is true; a dry distro would run afoul of the initial check and then just throw a 404.  But as you say, we don't have any dry distros anymore, so it's not really something I could test.  Should we just remove the code for dry distros?

tfoote commented:
> Should we just remove the code for dry distros?
I'd say yes to removing support for dry distros. We only support rosbuild based things from source in a workspace now and we don't intend to bring it back and don't expect anyone else to use it so clearing it makes sense.

clalancette committed:
Remove the DryDistro class.

We no longer need to do prerelease for dry distros, so just
remove that class and the one user.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

clalancette commented:
> I'd say yes to removing support for dry distros. We only support rosbuild based things from source in a workspace now and we don't intend to bring it back and don't expect anyone else to use it so clearing it makes sense.

Thanks.  Done in b62288e

sloretz commented:
ROS 1 is end-of-life (EOL) as of today, May 31st 2025. I am archiving this repository because it only supports ROS 1.

[If you still rely on ROS 1, read this page to learn about your options.](https://bit.ly/NoeticEOL)
